### PR TITLE
add comand to edit and refresh efpt.config.json files - 2

### PR DESCRIPTION
**Pull request by pedromcrocha**
Create a context menu in the file solution explorer as efpt. * Config.json
Edit configuration
Update

I made the necessary changes to ReverseEngineerHandler
More testing is needed with Dacpac (refresh)

### CODE REVIEW

**Comment by ErikEJ** [comment]
No files??

**Comment by pedromcrocha** [comment]
try now

File: src/GUI/EFCorePowerTools/EFCorePowerToolsPackage.cs [hunk 1]
```diff
@@ -143,6 +156,22 @@ protected override async System.Threading.Tasks.Task InitializeAsync(Cancellatio
...
+            await ThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
+
+            var menuCommand = sender as MenuCommand;
+            if (menuCommand == null || _dte2.SelectedItems.Count != 1)
+            {
+                return;
+            }
+
+            var itemName = _dte2.SelectedItems.Item(1).Name;
+            menuCommand.Visible = itemName.StartsWith("efpt.") && itemName.EndsWith("config.json");
```
Comment (ErikEJ) [inline src/GUI/EFCorePowerTools/EFCorePowerToolsPackage.cs:170]: Add null check and use StringComparision overload for StartsWith and EndsWith

File: src/GUI/EFCorePowerTools/EFCorePowerToolsPackage.cs [hunk 2]
```diff
@@ -173,6 +202,40 @@ private async System.Threading.Tasks.Task OnProjectMenuBeforeQueryStatusAsync(ob
...
+            await ThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
+
+            var menuCommand = sender as MenuCommand;
+            if (menuCommand == null || _dte2.SelectedItems.Count != 1)
+            {
+                return;
+            }
+
+            var itemName = _dte2.SelectedItems.Item(1).Name;
+            if (!itemName.StartsWith("efpt.") || !itemName.EndsWith("config.json"))
```
Comment (ErikEJ) [inline src/GUI/EFCorePowerTools/EFCorePowerToolsPackage.cs:216]: Same comment as above

**Comment by ErikEJ** [comment]
Thanks - I will check it out and have a look!

File: src/GUI/EFCorePowerTools/Handlers/ReverseEngineer/ReverseEngineerHandler.cs [hunk 3]
```diff
@@ -21,360 +21,461 @@ internal class ReverseEngineerHandler
...
+            {
+                var dataBaseInfo = databaseList.Values.FirstOrDefault(m => m.Caption == options.UiHint);
+                if (dataBaseInfo != null)
                 {
-                    EnvDteHelper.ShowError($"Unsupported provider: {dbInfo.ServerVersion}");
-                    return;
+                    options.ConnectionString = dataBaseInfo.ConnectionString;
+                    options.DatabaseType = dataBaseInfo.DatabaseType;
                 }
+            }
```
Comment (ErikEJ) [inline src/GUI/EFCorePowerTools/Handlers/ReverseEngineer/ReverseEngineerHandler.cs:159]: What if no connection is found? Either show a message, or run the wizard flow

File: src/GUI/EFCorePowerTools/Handlers/ReverseEngineer/ReverseEngineerHandler.cs [hunk 4]
```diff
@@ -21,360 +21,461 @@ internal class ReverseEngineerHandler
...
-                    var rightsAndVersion = reverseEngineerHelper.HasSqlServerViewDefinitionRightsAndVersion(options.ConnectionString);
+        private async System.Threading.Tasks.Task InstallEfCorePackageAsync(Project project, ReverseEngineerOptions options, Tuple<bool, string> containsEfCoreReference)
+        {
+            await ThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
 
-                    if (rightsAndVersion.Item1 == false)
-                    {
-                        EnvDteHelper.ShowMessage("The SQL Server user does not have 'VIEW DEFINITION' rights, default constraints may not be available.");
-                    }
+            if (containsEfCoreReference.Item1)
```
Comment (ErikEJ) [inline src/GUI/EFCorePowerTools/Handlers/ReverseEngineer/ReverseEngineerHandler.cs:370]: This should at least be inverted, but the old logic must be preserved if possible:

`if (modelingOptionsResult.Payload.InstallNuGetPackage)`

**Review by ErikEJ — CHANGES_REQUESTED** [review]

File: src/GUI/EFCorePowerTools/Handlers/ReverseEngineer/ReverseEngineerHandler.cs (hunk 4 above)
Comment (pedromcrocha) [reply src/GUI/EFCorePowerTools/Handlers/ReverseEngineer/ReverseEngineerHandler.cs:370]: please verify again, the method VerifySQLServerRightsAndVersion have the same code

Comment (pedromcrocha) [reply src/GUI/EFCorePowerTools/Handlers/ReverseEngineer/ReverseEngineerHandler.cs:370]: ok, i see now

Comment (ErikEJ) [reply src/GUI/EFCorePowerTools/Handlers/ReverseEngineer/ReverseEngineerHandler.cs]: I did a refresh, and the installed package was updated - basically Refresh should probably never install the provider package.

**Comment by ErikEJ** [comment]
I tested on a project with no efpt.config.json file, and got NRE on line 288:

if (options.Tables.Count > 0)

Should probably be: if (options.Tables?.Count > 0)

**Comment by ErikEJ** [comment]
Thank, this is a great contribution!

**Pull request merged**